src/Portable.Xaml/Portable.Xaml/XamlObjectWriter.cs

```diff
@@ -346,7 +346,6 @@ protected override void OnWriteStartObject()
 			if (!state.Type.IsContentValue(service_provider))
 				InitializeObjectIfRequired(true);
 			state.IsXamlWriterCreated = true;
-			source.OnBeforeProperties(state.Value);
```

Comment:
Hi @cm4ker! Because at this point, state.Value may be null. This event is now raised inside InitializeObjectIfRequired and PopulateObject, where we know for sure that state.Value isn't null (closer to HandleBeginInit).